### ISSUE TITLE
Trailing CR causes jspm to fail on Unix

### ISSUE THREAD

**Issue by sylvainr** [jspm/jspm-cli#2377]
**JSPM Version:** `beta.46`

**Transpiler Plugin(s):** I do not use a transpiler

**Details:**
It appears that some bad carriage return characters got their way into `jspm.js`. It just started happening with the new beta.

The consequence is that `beta.46` introduces issues on both Mac OS X and Linux, where the following issues shows up:

```
$ ./node_modules/.bin/jspm 
env: node\r: No such file or directory
```

The `\r` hints at some carriage return stuff. It is confirmed by running diff on `jspm.js` on the `beta-45` and `beta-46` with and without the `strip-trailing-cr` option:

```
$ diff --strip-trailing-cr /tmp/v45/node_modules/.bin/jspm /tmp/v46/node_modules/.bin/jspm | wc -l
       0
$ diff /tmp/v45/node_modules/.bin/jspm /tmp/v46/node_modules/.bin/jspm | wc -l
      88
```

**Comment by guybedford** [comment]
Thanks for posting. Which version of npm are you using here? It seems like this should be fixed for installs of newer versions.

**Comment by guybedford** [comment]
Reference: https://github.com/npm/npm/issues/12371

**Comment by sylvainr** [comment]
I am using `node 3.10.10` which is the most up to date version when installing Node LTS release ("Recommended for most users").

I still haver the issue.

**Comment by guybedford** [comment]
@sylvainr the recommended  NodeJS version is 6.11.4!

**Comment by sylvainr** [comment]
I am sorry, I meant `npm 3.10.10`.

**Comment by guybedford** [comment]
Ahh I see. I don't have access to a unix machine today, so if you're able to check running Node 8 with npm 5 let me know if that makes a difference.

**Comment by amenadiel** [comment]
I'm having the same issue, tho I can't see any CR and it doesn't happen in all of my environments. In my local, using node 8.x, there's no problem. In my AWS instances, running node 4.x, it does happen.

For the time being, I'm pinning `0.17-beta.45`.

@guybedford 
> Ahh I see. I don't have access to a unix machine today

This startled me. what is your usual dev environment?

**Comment by i8degrees** [comment]
Hi,

@guybedford This is happening for me, too, after upgrading to jspm v0.17.0-beta46 from jspm v0.17.0-beta42.

```bash
env: node\r: No such file or directory
```

Mac OS X "El Capitan" v10.11.6 (15G1217)
Node v8.4.0
npm v5.3.0

**Comment by havelock** [comment]
@guybedford We ran into this as well.

Our team is using jspm in a project and we ran into this both on Linux in CI/CD agent. One of the devs re-installing their environment on macOS experienced this, too.

**Comment by guybedford** [comment]
I've published a 0.17.0-beta.47 to resolve this. It's very odd that a standard NodeJS publish workflow doesn't work in Windows due to this convention breaking. Shows how unix the culture is!

**Comment by tjclement** [comment]
This seems to be reintroduced in 0.17.0-beta.48 again -> received the same error message on Mac OS, and downgrading to 47 resolved it.

**Comment by guybedford** [comment]
Fixed in 0.17.0-beta.49.

**Comment by torson** [comment]
I get the same `/usr/bin/env: ‘node\r’: No such file or directory` when using 0.16.54 , works with 0.16.53 .

**Comment by guybedford** [comment]
@torson thanks for reporting - that's a very annoying issue I should remember to avoid. I've published a 0.16.55 with the fix.
